OrangeHRMTests/Tests/Tests.cs

```diff
@@ -0,0 +1,562 @@
+
+            GenericPages.LeavePage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, newFirstName);
+            GenericPages.LeavePage.ClickDropdownListFirstPosition();
+            GenericPages.LeavePage.ChooseValueFromDropDownByName(DropDownFieldsNames.LeaveTypeDropDownFieldName, CANVacationDropDownFieldValue);
+            GenericPages.LeavePage.EnterValueInInputTextField(InputFieldsNames.EntitlementInputFieldName, NewEntitlement);
+            GenericPages.LeavePage.ClickSaveButton();
+            GenericPages.LeavePage.ClickConfirmButton();
+
+            GenericPages.LeavePage.TopBarMenu.ClickTopbarMenuButtonByName(LeavePageTopBarMenuButtonsNames.AssignLeaveTopBarMenuButtonName);
+
+            ClassicAssert.AreEqual(LeavePageTopBarMenuButtonsNames.AssignLeaveTopBarMenuButtonName, GenericPages.LeavePage.GetMainTitleText());
+
+            GenericPages.LeavePage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, newFirstName);
+            GenericPages.LeavePage.ClickDropdownListFirstPosition();
+            GenericPages.LeavePage.ChooseValueFromDropDownByName(DropDownFieldsNames.LeaveTypeDropDownFieldName, CANVacationDropDownFieldValue);
+
+            GenericPages.LeavePage.ClickDropDownListArrowButtonByName(DropDownFieldsNames.FromDateDropDownFieldName);
+            GenericPages.LeavePage.ClickFromDateCalendarValueButton(FromDateVacationValue);
+            GenericPages.LeavePage.ClickDropDownListArrowButtonByName(DropDownFieldsNames.ToDateDropDownFieldName);
+            GenericPages.LeavePage.ClickToDateCalendarValueButton(ToDateVacationValue);
+            GenericPages.LeavePage.ClickSaveButton();
+            //Implementation of wait while element appears - was discussed
+            string GetSuccessfullySavedMessageTextResult() => GenericPages.BasePage.GetPopUpMessageTextElement();
+
+            ClassicAssert.AreEqual(InfoMessageTextValues.SuccessfullySavedMessageText, GetSuccessfullySavedMessageTextResult());
+
+            GenericPages.BasePage.TopBarMenu.ClickTopbarMenuButtonByName(LeavePageTopBarMenuButtonsNames.LeaveListTopBarMenuButtonName);
+            GenericPages.LeavePage.ClickDropDownListArrowButtonByName(ShowLeaveWithStatusDropDownFieldValue);
+            GenericPages.LeavePage.ClickDropdownList(ScheduledDropDownFieldValue);
+            GenericPages.LeavePage.ChooseValueFromDropDownByName(DropDownFieldsNames.LeaveTypeDropDownFieldName, CANVacationDropDownFieldValue);
+            GenericPages.LeavePage.EnterValueInInputTextField(InputFieldsNames.EmployeeNameInputFieldName, newFirstName);
+            GenericPages.LeavePage.ClickDropdownListFirstPosition();
+            GenericPages.LeavePage.ClickSearchButton();
+
+            ClassicAssert.AreEqual(newFirstName + " " + newMiddleName + " " + newLastName, GenericPages.LeavePage.Table.GetCellText(TableColumnsNames.EmployeeNameTableColumnName));
```

Comment:
string.Join can help you with newFirstName + " " + newMiddleName + " " + newLastName
here and below as usual

Reply:
Fixed